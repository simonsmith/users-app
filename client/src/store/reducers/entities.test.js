import entitiesReducer from './entities';

test('should handle no state being passed', () => {
  expect(entitiesReducer(undefined, {})).toMatchSnapshot();
});

test('should merge the object into the state when the action payload contains an entities property', () => {
  const action = {
    payload: {
      entities: {
        users: {
          123: {name: 'bar'},
          456: {name: 'foo'},
        },
      },
    },
  };
  const beforeState = {
    users: {
      789: {name: 'baz'},
    },
    other: {foo: 'bar'},
  };
  const afterState = entitiesReducer(beforeState, action);

  expect(afterState).toMatchSnapshot();
  expect(afterState).not.toBe(beforeState);
});
