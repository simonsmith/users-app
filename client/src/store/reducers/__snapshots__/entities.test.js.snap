// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle no state being passed 1`] = `
Object {
  "users": Object {},
}
`;

exports[`should merge the object into the state when the action payload contains an entities property 1`] = `
Object {
  "other": Object {
    "foo": "bar",
  },
  "users": Object {
    "123": Object {
      "name": "bar",
    },
    "456": Object {
      "name": "foo",
    },
    "789": Object {
      "name": "baz",
    },
  },
}
`;
